battery-service/src/lib.rs

```diff
@@ -122,10 +128,24 @@ impl<
                     .await
                     .unwrap();
             }
+            BatteryMsgs::Oem(_) => (),
             _ => todo!(),
         }
         Ok(())
     }
+
+    pub async fn handle_new_power_policy(&self) {
+        let current = self.policy_signal.wait().await;
+        Timer::after_millis(1000).await;
+        let res = self
+            .charger
+            .rx
+            .try_send(BatteryMsgs::Oem(OemMessage::ChargeCurrent(current)));
+        if res.is_err() {
+            error!("Charger buffer full");
+        }
+        info!("SENT NEW CHARGE CURRENT");
```

Comment:
If the rx buffer is full, this will log both the error and info. Consider pattern matching to only log one or the other
```suggestion
        let res = self
            .charger
            .rx
            .try_send(BatteryMsgs::Oem(OemMessage::ChargeCurrent(current)));

        match res {
            Ok(_) => info!("SENT NEW CHARGE CURRENT"),
            Err(_) => error!("Charge buffer full"),
        }
```